common/plugins/eu.esdihumboldt.hale.common.align/src/eu/esdihumboldt/hale/common/align/model/AlignmentUtil.java

```diff
@@ -1045,13 +1046,23 @@ public static void addValues(Group group, List<ChildContext> path,
 
 				// check all values
 				for (Object value : values) {
-					if (value instanceof Group) {
+					if (value instanceof InstanceDecorator) {
+						if (((InstanceDecorator) value).getValue() != null) {
+							collectedValues.add(((InstanceDecorator) value).getValue());
+						}
+					}
+					else if (value instanceof Group) {
 						addValues((Group) value, subPath, collectedValues, onlyValues);
 					}
 					else if (subPath.isEmpty()) {
 						// normal value and at the end of the path
 						if (value != null) {
-							collectedValues.add(value);
+							if (value instanceof InstanceDecorator) {
```

Comment:
If the `subPath` is empty, then an `Instance` should actually be handled by the `value instanceof Group` condition above, where `addValues` is called again. There, with an empty path it then depends on the value of the parameter `onlyValues` if the `Instance` itself is added or its value extracted.